src/test/clojurescript/clara/test_salience.cljs

```diff
@@ -0,0 +1,115 @@
+(ns clara.test-salience
+  (:require-macros [cljs.test :refer (is deftest run-tests testing)]
+    [clara.test-rules-data])
+  (:require [cljs.test :as t]
+    [clara.rules.engine :as eng]
+    [clara.rules.accumulators :as acc]
+    [clara.rules :refer [insert retract fire-rules query insert!]
+     :refer-macros [defrule defsession defquery]]
+    [clara.rules.testfacts :refer [->Temperature Temperature
+                                   ->WindSpeed WindSpeed
+                                   ->ColdAndWindy ColdAndWindy]]))
+
+(def salience-rule-output (atom []))
+
+(defrule salience-rule1
+  {:salience 100}
+  [Temperature]
+  =>
+  (swap! salience-rule-output conj 100))
+
+(defrule salience-rule2
+  {:salience 50}
+  [Temperature]
+  =>
+  (swap! salience-rule-output conj 50))
+
+(defrule salience-rule3
+  {:salience 0}
+  [Temperature]
+  =>
+  (swap! salience-rule-output conj 0))
+
+(defrule salience-rule4
+  {:salience -50}
+  [Temperature]
+  =>
+  (swap! salience-rule-output conj -50))
+
+
+(deftest test-salience
+  (doseq [[sort-fn
+           group-fn
+           expected-order]
+
+          [[:default-sort :default-group :forward-order]
+           [:default-sort :salience-group :forward-order]
+           [:default-sort :neg-salience-group :backward-order]
+
+           [:numeric-greatest-sort :default-group :forward-order]
+           [:numeric-greatest-sort :salience-group :forward-order]
+           [:numeric-greatest-sort :neg-salience-group :backward-order]
+
+
+           [:boolean-greatest-sort :default-group :forward-order]
+           [:boolean-greatest-sort :salience-group :forward-order]
+           [:boolean-greatest-sort :neg-salience-group :backward-order]
+
+
+           [:numeric-least-sort :default-group :backward-order]
+           [:numeric-least-sort :salience-group :backward-order]
+           [:numeric-least-sort :neg-salience-group :forward-order]
+
+           [:boolean-least-sort :default-group :backward-order]
+           [:boolean-least-sort :salience-group :backward-order]
+           [:boolean-least-sort :neg-salience-group :forward-order]]]
+
+    (let [numeric-greatest-sort (fn [x y]
+                                  (cond
+                                    (= x y) 0
+                                    (> x y) -1
+                                    :else 1))
+
+          numeric-least-sort (fn [x y]
+                               (numeric-greatest-sort y x))
+
+          salience-group-fn (fn [production]
+                              (or (some-> production :props :salience)
+                                0))
+
+          neg-salience-group-fn (fn [p]
+                                  (- (salience-group-fn p)))]
+
+      (defsession test-salience-session 'clara.test-salience
```

Comment:
On reflection, I think this is OK for the moment.  It is a bit odd, but I don't foresee the structure of the defsession macro changing in the immediate future in a way that would break this usage.

It does bring up the interesting point of whether we should expose a macro like

````
(mk-compile-time-session rule-sources :option option-value)
````

in ClojureScript that would behave like this without the requirement of defining a var, either as a test helper internal to Clara's tests or as a public API.  It wouldn't be too difficult to refactor the [defsession implementation](https://github.com/cerner/clara-rules/blob/0.14.0/src/main/clojure/clara/macros.clj#L251) to share the logic in that way.  Alternatively, perhaps eval in ClojureScript has matured to the point that we could use it in our tests.  I don't think we should make decisions on these points in haste, so I'm in favor of leaving the defsession usage in these tests as is.   We should probably have an issue to address these points; I'll plan to log an issue (or you can) and then we can add a comment next to the defsession use here referencing that issue.